### PR TITLE
#1595 Update Recurring Donation Paid Amount when related Closed/Won O…

### LINKED ISSUE #1595: Recurring Donation child opportunities should roll up Paid Amount when inserted as Closed/Won, not just on update to Closed/Won
Hub thread: https://powerofus.force.com/0D580000029U9Ui

### PR DESCRIPTION
**Pull request by kenslewis**
…pp is inserted and prevents Divide by 0 error if Refresh Opps button is clicked when number of installments is equal to number of paid installments.

# Warning
# Info
Users/Admins will have to update the Trigger_Handler__c record for the RD_RecurringDonations_Opp_TDTM class to include the 'AfterInsert' trigger action.
# Issues 
Does not change how the npe03__Recurring_Donation__c. npe03__Installment_Amount__c field is calculated. So open ended recurring donations with Installments equal to Total Paid Installments will show an Installment Amount equal to the Amount, even if Paid Amount is greater than 0. 

Fixed #1595

### CODE REVIEW

File: src/classes/RD_RecurringDonations.cls [hunk 1]
```diff
@@ -386,8 +386,9 @@ public with sharing class RD_RecurringDonations {
                 if (r.npe03__Total_Paid_Installments__c != null) {
                 	paidInstallments = integer.valueOf(r.npe03__Total_Paid_Installments__c);
                     j = paidInstallments;
-                    if (r.npe03__Schedule_Type__c != Label.npe03.RecurringDonationMultiplyValue) {                    
-                        installmentAmount = (r.npe03__Amount__c - paidAmount) / (installments - paidInstallments);
+                    if (r.npe03__Schedule_Type__c != Label.npe03.RecurringDonationMultiplyValue) {
+                        if (installments - paidInstallments > 0)
```
Comment (kenslewis) [inline src/classes/RD_RecurringDonations.cls]: Prevent 'Divide by 0' error by first checking that installments - paidInstallments > 0

**Comment by njjc** [comment]
Great work and thanks for the contribution, @kenslewis. I'll be testing and merging this in shortly, it will make it into our next release.

**Comment by kenslewis** [comment]
Great! Thanks, @njjc.

**Comment by njjc** [comment]
Per your info, the post install/upgrade script covers the modification of the TDTM trigger handler record. 

Pull request moved to https://github.com/SalesforceFoundation/Cumulus/pull/1612 to run through build system.

**Pull request closed without merging**